#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: This dpatch addes many locales especially for UTF-8 and 
# DP:		ISO-8859-15, to display debconf locale menu using
# DP:		SUPPORTED.orig. This dpatch is based on RedHat cygnus-to-
# DP:		redhat patch.
# DP: Author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Debian/RedHat-Specific
# DP: Status Details: Severely reduced because many of these locales
#                     are now in upstream. - jbailey 2003-03-23
# DP: Date: 2003-02-09, Update 2004-07-21 for #246170 - gotom

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- localedata/SUPPORTED~	2004-07-21 09:25:57.000000000 +0900
+++ localedata/SUPPORTED	2004-07-21 09:26:34.000000000 +0900
@@ -56,6 +56,7 @@
 bs_BA/ISO-8859-2 \
 byn_ER/UTF-8 \
 ca_ES.UTF-8/UTF-8 \
+ca_ES.UTF-8@euro/UTF-8 \
 ca_ES/ISO-8859-1 \
 ca_ES@euro/ISO-8859-15 \
 cs_CZ.UTF-8/UTF-8 \
@@ -64,18 +65,23 @@
 cy_GB/ISO-8859-14 \
 da_DK.UTF-8/UTF-8 \
 da_DK/ISO-8859-1 \
+da_DK.ISO-8859-15/ISO-8859-15 \
 de_AT.UTF-8/UTF-8 \
+de_AT.UTF-8@euro/UTF-8 \
 de_AT/ISO-8859-1 \
 de_AT@euro/ISO-8859-15 \
 de_BE.UTF-8/UTF-8 \
+de_BE.UTF-8@euro/UTF-8 \
 de_BE/ISO-8859-1 \
 de_BE@euro/ISO-8859-15 \
 de_CH.UTF-8/UTF-8 \
 de_CH/ISO-8859-1 \
 de_DE.UTF-8/UTF-8 \
+de_DE.UTF-8@euro/UTF-8 \
 de_DE/ISO-8859-1 \
 de_DE@euro/ISO-8859-15 \
 de_LU.UTF-8/UTF-8 \
+de_LU.UTF-8@euro/UTF-8 \
 de_LU/ISO-8859-1 \
 de_LU@euro/ISO-8859-15 \
 el_GR.UTF-8/UTF-8 \
@@ -90,9 +96,11 @@
 en_DK/ISO-8859-1 \
 en_GB.UTF-8/UTF-8 \
 en_GB/ISO-8859-1 \
+en_GB.ISO-8859-15/ISO-8859-15 \
 en_HK.UTF-8/UTF-8 \
 en_HK/ISO-8859-1 \
 en_IE.UTF-8/UTF-8 \
+en_IE.UTF-8@euro/UTF-8 \
 en_IE/ISO-8859-1 \
 en_IE@euro/ISO-8859-15 \
 en_IN/UTF-8 \
@@ -104,6 +112,7 @@
 en_SG/ISO-8859-1 \
 en_US.UTF-8/UTF-8 \
 en_US/ISO-8859-1 \
+en_US.ISO-8859-15/ISO-8859-15 \
 en_ZA.UTF-8/UTF-8 \
 en_ZA/ISO-8859-1 \
 en_ZW.UTF-8/UTF-8 \
@@ -123,6 +132,7 @@
 es_EC.UTF-8/UTF-8 \
 es_EC/ISO-8859-1 \
 es_ES.UTF-8/UTF-8 \
+es_ES.UTF-8@euro/UTF-8 \
 es_ES/ISO-8859-1 \
 es_ES@euro/ISO-8859-15 \
 es_GT.UTF-8/UTF-8 \
@@ -152,15 +162,18 @@
 et_EE.UTF-8/UTF-8 \
 et_EE/ISO-8859-1 \
 eu_ES.UTF-8/UTF-8 \
+eu_ES.UTF-8@euro/UTF-8 \
 eu_ES/ISO-8859-1 \
 eu_ES@euro/ISO-8859-15 \
 fa_IR/UTF-8 \
 fi_FI.UTF-8/UTF-8 \
+fi_FI.UTF-8@euro/UTF-8 \
 fi_FI/ISO-8859-1 \
 fi_FI@euro/ISO-8859-15 \
 fo_FO.UTF-8/UTF-8 \
 fo_FO/ISO-8859-1 \
 fr_BE.UTF-8/UTF-8 \
+fr_BE.UTF-8@euro/UTF-8 \
 fr_BE/ISO-8859-1 \
 fr_BE@euro/ISO-8859-15 \
 fr_CA.UTF-8/UTF-8 \
@@ -168,12 +181,15 @@
 fr_CH.UTF-8/UTF-8 \
 fr_CH/ISO-8859-1 \
 fr_FR.UTF-8/UTF-8 \
+fr_FR.UTF-8@euro/UTF-8 \
 fr_FR/ISO-8859-1 \
 fr_FR@euro/ISO-8859-15 \
 fr_LU.UTF-8/UTF-8 \
+fr_LU.UTF-8@euro/UTF-8 \
 fr_LU/ISO-8859-1 \
 fr_LU@euro/ISO-8859-15 \
 ga_IE.UTF-8/UTF-8 \
+ga_IE.UTF-8@euro/UTF-8 \
 ga_IE/ISO-8859-1 \
 ga_IE@euro/ISO-8859-15 \
 gd_GB/ISO-8859-15 \
@@ -182,6 +198,7 @@
 gez_ET/UTF-8 \
 gez_ET@abegede/UTF-8 \
 gl_ES.UTF-8/UTF-8 \
+gl_ES.UTF-8@euro/UTF-8 \
 gl_ES/ISO-8859-1 \
 gl_ES@euro/ISO-8859-15 \
 gu_IN/UTF-8 \
@@ -201,6 +218,7 @@
 it_CH.UTF-8/UTF-8 \
 it_CH/ISO-8859-1 \
 it_IT.UTF-8/UTF-8 \
+it_IT.UTF-8@euro/UTF-8 \
 it_IT/ISO-8859-1 \
 it_IT@euro/ISO-8859-15 \
 iw_IL.UTF-8/UTF-8 \
@@ -233,11 +251,15 @@
 mt_MT/ISO-8859-3 \
 nb_NO.UTF-8/UTF-8 \
 nb_NO/ISO-8859-1 \
+no_NO.UTF-8/UTF-8 \
+no_NO/ISO-8859-1 \
 ne_NP/UTF-8 \
 nl_BE.UTF-8/UTF-8 \
+nl_BE.UTF-8@euro/UTF-8 \
 nl_BE/ISO-8859-1 \
 nl_BE@euro/ISO-8859-15 \
 nl_NL.UTF-8/UTF-8 \
+nl_NL.UTF-8@euro/UTF-8 \
 nl_NL/ISO-8859-1 \
 nl_NL@euro/ISO-8859-15 \
 nn_NO.UTF-8/UTF-8 \
@@ -251,6 +273,7 @@
 pt_BR.UTF-8/UTF-8 \
 pt_BR/ISO-8859-1 \
 pt_PT.UTF-8/UTF-8 \
+pt_PT.UTF-8@euro/UTF-8 \
 pt_PT/ISO-8859-1 \
 pt_PT@euro/ISO-8859-15 \
 ro_RO.UTF-8/UTF-8 \
@@ -279,10 +302,12 @@
 st_ZA.UTF-8/UTF-8 \
 st_ZA/ISO-8859-1 \
 sv_FI.UTF-8/UTF-8 \
+sv_FI.UTF-8@euro/UTF-8 \
 sv_FI/ISO-8859-1 \
 sv_FI@euro/ISO-8859-15 \
 sv_SE.UTF-8/UTF-8 \
 sv_SE/ISO-8859-1 \
+sv_SE.ISO-8859-15/ISO-8859-15 \
 ta_IN/UTF-8 \
 te_IN/UTF-8 \
 tg_TJ/KOI8-T \
